compiler/arser/include/arser/arser.h

```diff
@@ -119,10 +146,41 @@ enum class DataType
 
 class Arser;
 
+/**
+ * Argument
+ *   ├── positional argument
+ *   └── optioanl argument  [ dash at the beginning of the string ]
+ *       ├── long option    [ two or more dashes ]
+ *       └── short option   [ one dash ]
+ *
+ * Argument has two types - positional argument, optional argument.
+ *
+ * The way to distinguish the two types is whether there is a dash('-') at the beginning of the
+ * string.
+ *
+ * And, optional argument has two types as well - long option, short option, which is distinguished
```

Comment:
What I assumed is, long option with full name, like `--help`and short option with one dash and one character like `-h`
from this comment, I has some questions like above...

Reply:
ditto. That's the basic assumption. Even though user gives `----h` or etc, it will work.